ostd/src/task/processor.rs

```diff
@@ -99,10 +109,14 @@ pub(super) fn switch_to_task(next_task: Arc<Task>) {
     // next task. Not dropping is just fine because the only consequence is that we delay the drop
```

Comment:
In light of this PR where the pointer to the previous task, I think we should rethink this design decision. After context switching to the next task, we should always recover the `Arc<Task>` to the previous task and do the following:
1. Reset the foreground flag of the previous one;
2. Call the post scheduler callback handler;
3. Drop the previous task.